jaxrs-tck/src/main/java/ee/jakarta/tck/ws/rs/jaxrs31/ee/multipart/MultipartSupportIT.java

```diff
@@ -0,0 +1,427 @@
+                "  </mid>" + LS +
+                "  <mid attr1=\"value4\" attr2=\"value5\">" + LS +
+                "    <inner attr3=\"value6\"/>" + LS +
+                "  </mid>" + LS +
+                "</root>";
+        return new ByteArrayInputStream(xml.getBytes());
+    }
+
+
+    @BeforeEach
+    void logStartTest(TestInfo testInfo) {
+        TestUtil.logMsg("STARTING TEST : "+testInfo.getDisplayName());
+    }
+
+    @AfterEach
+    void logFinishTest(TestInfo testInfo) {
+        TestUtil.logMsg("FINISHED TEST : "+testInfo.getDisplayName());
+    }
+
+    @Deployment
+    public static WebArchive deployment() {
+        return ShrinkWrap.create(WebArchive.class, MultipartSupportIT.class.getSimpleName() + ".war")
+                .addAsWebInfResource(EmptyAsset.INSTANCE, "beans.xml");
+    }
+
+    @ArquillianResource
+    private URI uri;
+
+    /**
+     * Verify sending a List of {@link EntityPart} and receiving a List sent back from the resource method containing
+     * the same parts plus an additional part added by thye resource method containing extra headers.
+     *
+     * @throws Exception if an error occurs in the test
+     */
+    @Test
+    public void basicTest() throws Exception {
+        try (Client client = ClientBuilder.newClient()) {
+            final List<EntityPart> multipart = List.of(
+                    EntityPart.withName("octet-stream")
+                            .content("test string".getBytes(StandardCharsets.UTF_8))
+                            .mediaType(MediaType.APPLICATION_OCTET_STREAM_TYPE)
+                            .build(),
+                    EntityPart.withName("file")
+                            .content("test file", xmlFile())
+                            .mediaType(MediaType.APPLICATION_XML)
+                            .build()
+                    );
+            try (
+                    Response response = client.target(createCombinedUri(uri, "test/basicTest"))
+                            .request(MediaType.MULTIPART_FORM_DATA_TYPE)
+                            .post(Entity.entity(new GenericEntity<>(multipart) {
+                            }, MediaType.MULTIPART_FORM_DATA))) {
+                Assertions.assertEquals(200, response.getStatus());
+                final List<EntityPart> entityParts = response.readEntity(new GenericType<>() {
+                });
+                if (entityParts.size() != 3) {
+                    final String msg = "Expected 3 entries, received " +
+                            entityParts.size() +
+                            '.' +
+                            System.lineSeparator() +
+                            getMessage(entityParts);
+                    Assertions.fail(msg);
+                }
+                EntityPart part = find(entityParts, "received-string");
+                Assertions.assertNotNull(part, getMessage(entityParts));
+                Assertions.assertEquals("test string", part.getContent(String.class));
```

Comment:
The javadoc of getContent() says:
> Subsequent invocations will result in an IllegalStateException. Likewise this method will throw an IllegalStateException if it is called after calling getContent or getContent(GenericType).

But the getContent() has already been called by getMessage(). It looks like RestEasy allows for reading getContent() multiple times. Does it make sense to add a negative test to check the getContent has not been invoked multiple times?

Reply:
Good catch

Reply:
Yes, good catch.  I will fix the test and your suggestion to add a negative test is appropriate.

Reply:
@jansupol Looking at the  [getContent javadoc](https://jakarta.ee/specifications/restful-ws/3.1/apidocs/jakarta.ws.rs/jakarta/ws/rs/core/entitypart#getContent()), shouldn't the no-parameter method also throw IllegalStateException since it could be called twice or after one of the other overloads?

Reply:
I'd say not necessarily, since it returns the InputStream (for instance the ServletInputStream on the server side), which is meant to be the native datatype of the wired data, so the InputStream can be returned multiple times, although it can be already consumed, so the stream is not expected to be consumed twice, unlike with converting it to other data types (such as String) which are not supposed to be empty.

Reply:
But sure, there is a discrepancy to getContent(InputStream.class) which throws, so I would not mind the no-arg method to throw, too. Or the InputStream can be excluded from throwing the ISE. Either way is possible.

Reply:
The changes I've made this afternoon resolve this issue and pass on RESTEasy after @jamezp made the appropriate changes.